python/ray/tune/trainable/trainable.py

```diff
@@ -864,6 +864,27 @@ def restore(
                 could not be found.
 
         """
+        if _use_storage_context():
+            from ray.train._internal.checkpoint_manager import _TrainingResult
+
+            checkpoint_result = checkpoint_path
+            assert isinstance(checkpoint_result, _TrainingResult)
+
+            # TODO(justinvyu): The Trainable `load_checkpoint` interface
+            # should be updated to take in a `_TrainingResult` / Checkpoint
+            self.load_checkpoint(checkpoint_result)
+
+            # TODO(justinvyu): What should we do about Trainable metadata?
```

Comment:
What's the plan for this?

Reply:
The `result.json` only keeps metrics and doesn't store which metrics are tied to a certain checkpoint number. Driver memory just holds the latest checkpoint / metrics for each trial.

Another thing is that users can technically disable all logging to `result.json`, so this metadata is in the checkpoint folder so that it can be self-contained.

Reply:
Ok, able to restore from metrics now.

There's only one other blocker for removing the tune metadata file completely, that I can address in a follow-up PR:
- We have a `Result.from_path(...)` API that reads a result from a trial folder.
    - It assumes no access to the experiment level dir / Trial data.
    - Result needs to load checkpoints + best checkpoints and their results. So, this may require adding a "checkpoint_index" field to entries in the result.json that correspond to a checkpoint.

Reply:
Yeah seems fine to address in a follow-up. Result.from_path() is also a standalone util, so I'm a bit less concerned about it doing potentially sketchy things like reading the JSON file.